src/domain/events/EventsPage.tsx

```diff
@@ -1,19 +1,140 @@
-import { Button } from 'hds-react';
+import { Button, TextInput } from 'hds-react';
 import React from 'react';
 import { useTranslation } from 'react-i18next';
 import { useHistory } from 'react-router';
 
+import LoadingSpinner from '../../common/components/loadingSpinner/LoadingSpinner';
+import {
+  EventFieldsFragment,
+  EventsQuery,
+  useEventsQuery,
+} from '../../generated/graphql';
+import useDebounce from '../../hooks/useDebounce';
+import useLocale from '../../hooks/useLocale';
+import getLocalizedString from '../../utils/getLocalizedString';
+import getPageFromUrl from '../../utils/getPageFromUrl';
 import Container from '../app/layout/Container';
 import PageWrapper from '../app/layout/PageWrapper';
 import { ROUTES } from '../app/routes/constants';
+import EventCard from '../event/eventCard/EventCard';
+import { hasComingOccurrences, hasOccurrences } from '../event/utils';
+import { EVENT_SORT_KEYS, PAGE_SIZE } from './constants';
 import styles from './eventsPage.module.scss';
 
+const EventsTitle: React.FC<{ count: number; title: string }> = ({
+  count,
+  title,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <h2>
+      {title}{' '}
+      <span className={styles.eventCount}>
+        {t('events.textEventCount', {
+          count: count,
+        })}
+      </span>
+    </h2>
+  );
+};
+
+const Events: React.FC<{
+  events: EventFieldsFragment[];
+  goToEventOccurrencesPage: (id: string) => void;
+}> = ({ events, goToEventOccurrencesPage }) => {
+  const locale = useLocale();
+
+  return (
+    <div className={styles.eventsContainer}>
+      {events?.map((event) => {
+        return (
+          <EventCard
+            description={getLocalizedString(event.description || {}, locale)}
+            enrolmentsCount={0}
+            id={event.id || ''}
+            image={event.images[0]?.url}
+            name={getLocalizedString(event.name || {}, locale)}
+            occurrencesCount={event.pEvent?.occurrences.edges.length || 0}
+            onClick={goToEventOccurrencesPage}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
 const EventsPage = () => {
+  const [inputValue, setInputValue] = React.useState('');
+  const searchValue = useDebounce(inputValue, 100);
   const { t } = useTranslation();
+  const locale = useLocale();
   const history = useHistory();
+  const [isLoadingMore, setIsLoadingMore] = React.useState(false);
+
+  const { data, loading, fetchMore } = useEventsQuery({
+    errorPolicy: 'ignore',
```

Comment:
Does API return error here in some situations? Maybe backend shouldn't return error in some situations that it does now

Reply:
Yes, we are now creating events locally and to http://palvelutarjotin-api.test.kuva.hel.ninja/. If pEvent doesn't exist in used environment Apollo will throw an error without errorPolicy: "ignore"